### PR TITLE
verify User pw generation and creating w/ just email

### PR DESCRIPTION
**Pull request by gordonleacock**
This is part of ticket #211

### CODE REVIEW

**Comment by gordonleacock** [comment]
I'm using stub_model for all but one test (the User.create with email one).

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1,5 +1,21 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+  context "user can generate a password" do
+    let! (:a_user) {stub_model(User, email: "name1@blah.com")}
+    it "receives .generate_password and returns a password" do
+      allow(a_user).to receive(:generate_password).and_return("12345")
+      expect(a_user.generate_password).to eq("12345")
```
Comment (grosscol) [inline spec/models/user_spec.rb]: This is testing the mock.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+    end
+    it "receives .password_confirmation and matches the password" do
+      allow(a_user).to receive(:password).and_return("12345")
+      allow(a_user).to receive(:password_confirmation).and_return("12345")
+      expect(a_user.password_confirmation).to eq(a_user.password)
```
Comment (grosscol) [inline spec/models/user_spec.rb]: This is also testing the mock.

**Comment by grosscol** [comment]
The generate_password function is there to automatically fill in a password and password confirmation when create is called so Devise's model validation is satisfied.

Checking the function of `generate_password` can be done by making a `new` User, calling `generate_password` and then checking the results.  You can try this in the rails console by:
```
u = User.new(email: 'user@example.com')
puts u.password
puts u.password_confirmation
u.generate_password
puts u.password
puts u.password_confirmation
```

If you want to see the process fail as it did before `generate_password`, patch the generate_password method of a user instance to be a no-op and then try to save it.

**Comment by gordonleacock** [comment]
Thanks @grosscol, I've updated the spec as you describe and removed the mocks.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
-end
+  context "when a user is created it" do
+    it "can be created with just an email parameter" do
+      u1 = User.create(email: "name@blah.com")
+      expect(u1).to be_an_instance_of(User)
+      expect(u1.email).to eq("name@blah.com")
+    end
+    it "can generate a password" do
+      u2 = User.create(email: "name2@blah.com")
+      u2.generate_password
```
Comment (grosscol) [inline spec/models/user_spec.rb]: Calling `.create` already invokes `generate_password`.  To test `generate_password` explicitly like this, use `User.new` which does not make that call for you.

If you really want to be pedantic about it, you could `expect(u2.password).to be_nil` prior to the `u2.generate_password` line.

**Pull request merged**